.github/workflows/lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Code Quality Checks
+
+on: [push]
```

Comment:
shall we limit this to on push on master and on pull request against master? I think we don't need to run for any branch push unless there's a PR